### PR TITLE
fix: commands ignore --usetoolingapi

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
fix casing to honor the flag `usetoolingapi`

### What issues does this PR fix or reference?
@W-8879900@

### CODE REVIEW

**Review by peternhale — APPROVED** [review]

File: packages/plugin-data/src/dataCommand.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export abstract class DataCommand extends SfdxCommand {
   }
 
   public getConnection(): BaseConnection {
-    const connection: BaseConnection & ConnectionInternals = this.flags.useToolingApi
+    const connection: BaseConnection & ConnectionInternals = this.flags.usetoolingapi
```
Comment (WillieRuemmele) [inline packages/plugin-data/src/dataCommand.ts:97]: do all commands that have that flag extend `dataCommand` and use this `getConnection` method?

Comment (peternhale) [reply packages/plugin-data/src/dataCommand.ts:97]: @WillieRuemmele create/update/delete/get/query use the flag usertoolingapi

Comment (mshanemc) [reply packages/plugin-data/src/dataCommand.ts:97]: yes, they do.

Comment (WillieRuemmele) [reply packages/plugin-data/src/dataCommand.ts:97]: yea, but a few weeks ago `soql:query` didn't extend `DataCommand` and used `this.org.getConnection` not `this.getConnection` I just want to make sure this code is being called for all commands that could use it

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**